OptimizelySDK/Utils/CustomAttributeConditionEvaluator.cs

```diff
@@ -0,0 +1,46 @@
+﻿/* 
+ * Copyright 2019, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Newtonsoft.Json.Linq;
+using OptimizelySDK.Entity;
+
+namespace OptimizelySDK.Utils
+{
+    public static class CustomAttributeConditionEvaluator
+    {
+        /// <summary>
+        /// String constant representing custome attribute condition type.
+        /// </summary>
+        public const string CUSTOM_ATTRIBUTE_CONDITION_TYPE = "custom_attribute";
+
+        public static bool? Evaluate(JToken condition, UserAttributes userAttributes)
```

Comment:
Would there be any value in mapping the `condition` to and Optimizely `condition` class instead of passing around the JToken?

Reply:
That's a nice idea but this seems like a major change. Do you want it to be done here in this PR?

Reply:
No, let's punt on it for now.